util/util.go

```diff
@@ -170,3 +182,87 @@ func SliceRemoveIfPresent(list []interface{}, search interface{}) []interface{}
 
 	return list
 }
+
+// Example data:
+//   - userSuppliedPath = "transform"
+//   - endpoint = "/transform/role/{name}"
+//   - parameters will include path parameters
+func ParsePath(userSuppliedPath, endpoint string, d *schema.ResourceData) string {
+	fields := strings.Split(endpoint, "/")
+	// The second field should be the one the user supplied rather
+	// than the default one shown.
+	fields[1] = userSuppliedPath
+
+	// Since endpoints start with a "/", the first field is always
+	// an extraneous "" and should be dropped.
+	recomprised := "/" + strings.Join(fields[1:], "/")
+
+	// For a recomprised string like "/my-transform/role/{name}",
+	// this will return the fields of "/transform/role/" and
+	// "name".
+	fields = strings.FieldsFunc(recomprised, func(c rune) bool {
+		return c == '{' || c == '}'
+	})
+	for _, field := range fields {
+		valRaw, ok := d.GetOk(field)
+		if !ok {
+			continue
+		}
+		// All path parameters must be strings so it's safe to
+		// assume here.
+		val := valRaw.(string)
+		recomprised = strings.Replace(recomprised, fmt.Sprintf("{%s}", field), val, -1)
+	}
+	return recomprised
+}
+
+// For an endpoint like "/transform/role/{name}", returns
+// "{name}".
+func LastField(endpoint string) string {
+	endpointFields := strings.Split(endpoint, "/")
+	lastFieldPosition := len(endpointFields) - 1
+	if lastFieldPosition < 0 {
+		lastFieldPosition = 0
+	}
+	return endpointFields[lastFieldPosition]
+}
+
+// PathParameters is just like regexp FindStringSubmatch,
+// but it validates that the match is different from the string passed
+// in, and that there's only one result.
+func PathParameters(endpoint, vaultPath string) (map[string]string, error) {
+	fields := strings.Split(endpoint, "/")
+	fields[1] = "{path}"
+
+	for i, field := range fields {
+		if strings.HasPrefix(field, "{") {
+			fields[i] = strings.ReplaceAll(fields[i], "{", "(?P<")
+			fields[i] = strings.ReplaceAll(fields[i], "}", ">.+)")
+		}
+	}
+	pattern := strings.Join(fields, "/")
+
+	endpointReg, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
```

Comment:
nit: `fmt.Errorf("unable to compile regex: %q: %w\n", pattern, err)`

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/746/commits/7b97a2958ded16a63be8eceda55b03450156ffe5